src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -1883,5 +1885,457 @@ public static void Sort<T>(this Span<T> span, Comparison<T> comparison)
+                return new InterpolatedTryWriteBuilder(destination, provider, success);
+            }
+
+            /// <summary>Writes the specified string to the builder.</summary>
+            /// <param name="value">The string to write.</param>
+            /// <returns>true if the value could be formatted to the span; otherwise, false.</returns>
+            public bool AppendLiteral(string value)
+            {
+                if (value.TryCopyTo(_destination.Slice(_pos)))
+                {
+                    _pos += value.Length;
+                    return true;
+                }
+
+                return Fail();
+            }
+
+            #region AppendFormatted
+            // Design note:
+            // This provides the same set of overloads and semantics as InterpolatedStringBuilder.
+
+            #region AppendFormatted T
+            /// <summary>Writes the specified value to the builder.</summary>
+            /// <param name="value">The value to write.</param>
+            public bool AppendFormatted<T>(T value)
+            {
+                // This method could delegate to AppendFormatted with a null format, but explicitly passing
+                // default as the format to TryFormat helps to improve code quality in some cases when TryFormat is inlined,
+                // e.g. for Int32 it enables the JIT to eliminate code in the inlined method based on a length check on the format.
+
+                // If there's a custom formatter, always use it.
+                if (_hasCustomFormatter)
+                {
+                    return AppendCustomFormatter(value, format: null);
+                }
+
+                // Check first for IFormattable, even though we'll prefer to use ISpanFormattable, as the latter
+                // derives from the former.  For value types, it won't matter as the type checks devolve into
+                // JIT-time constants.  For reference types, they're more likely to implement IFormattable
+                // than they are to implement ISpanFormattable: if they don't implement either, we save an
+                // interface check over first checking for ISpanFormattable and then for IFormattable, and
+                // if it only implements IFormattable, we come out even: only if it implements both do we
+                // end up paying for an extra interface check.
+                string? s;
+                if (value is IFormattable)
+                {
+                    // If the value can format itself directly into our buffer, do so.
+                    if (value is ISpanFormattable)
+                    {
+                        int charsWritten;
+                        if (((ISpanFormattable)value).TryFormat(_destination.Slice(_pos), out charsWritten, default, _provider)) // constrained call avoiding boxing for value types
+                        {
+                            _pos += charsWritten;
+                            return true;
```

Comment:
In `StringBuilder` you were defensive against bad TryFormat implementations reporting they wrote out of bounds, but here you weren't.

Reply:
I did so in StringBuilder because it uses unsafe code that might trust the values in its length fields.  Here we just index into spans.